### ISSUE TITLE
Version 2.6 is broken

### ISSUE THREAD

**Issue by vhiairrassary** [DataDog/java-dogstatsd-client#52]
Hello, using 2.5 everything is fine, but when updating to 2.6 then our dashboards are empty. Nothing seems to be received by Datadog. We are trying to find where is the regression.

No exception are logged using `StatsDClientErrorHandler`.
How can we help you to troubleshoot this?

**Comment by vhiairrassary** [comment]
Locally, using a debug server* data is correctly received, but when switching to 2.6 nothing is received.

\* server code:
```scala
import java.io.IOException
import java.net.InetSocketAddress
import java.nio.ByteBuffer
import java.nio.channels.DatagramChannel
import java.nio.charset.StandardCharsets

final class DummyStatsDServer {
  private var server: DatagramChannel = _

  def this(port: Int) {
    this()
    server = DatagramChannel.open
    server.bind(new InetSocketAddress(port))
    this.listen()
  }

  private def listen(): Unit = {
    val thread = new Thread(new Runnable() {
      override def run(): Unit = {
        val packet = ByteBuffer.allocate(1500)
        while ( {
          server.isOpen
        }) if (false) try
          Thread.sleep(10)
        catch {
          case e: InterruptedException =>
            e.printStackTrace()
        }
        else try {
          packet.clear
          server.receive(packet)
          packet.flip
          for (msg <- StandardCharsets.UTF_8.decode(packet).toString.split("\n")) {
            println(s"New message: $msg")
          }
        } catch {
          case e: IOException =>
            e.printStackTrace()
        }
      }
    })
    thread.setDaemon(true)
    thread.start()
  }
}

object Main extends App {
  new DummyStatsDServer(8125)

  while(true) {
    Thread.sleep(1000L)
  }
}
```

**Comment by vhiairrassary** [comment]
An exception is triggered by `sendBuffer.flip();` (https://github.com/DataDog/java-dogstatsd-client/blob/master/src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java#L1043):

Exception message: java nio Bytebuffer no such method error

```
java -version
java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode)
```

**Comment by 2ndalpha** [comment]
Please fix it or remove the v2.6.
I just wasted 5 hours trying to get statsd working first time.

**Comment by nmuesch** [comment]
Hey, thanks for the report and apologies for the inconvenience. I'm going to try to reproduce with the server sample provided by @vhiairrassary 

@2ndalpha While we work on addressing this issue, could you pin your dependency to 2.5? Apologies for the inconvenience here.

**Comment by jfernandez** [comment]
I can confirm that v2.6 is **not** sending metrics to statsd. Working fine in 2.5.

I also strongly suggest you remove v2.6, it will affect your first time user experience (like for us).

**Comment by nmuesch** [comment]
Hey! I've been working to reproduce and isolate the issue here. It appears this was due to an issue with the build environment using a newer version of Java to compile the library for release. I'm hoping to cut a 2.6.1 bugfix tomorrow and will post back here when thats done (should be quicker than attempting to remove the 2.6 jar). Thanks and apologies again for the inconvenience.

**Comment by nmuesch** [comment]
@vhiairrassary Version 2.6.1 was released this morning and should address the issue you faced. Could you upgrade to this version and confirm things are working as expected for you? Thanks for the report and detailed information and apologies again for the inconvenience.

**Comment by vhiairrassary** [comment]
@nmuesch Awesome 🎉! Version 2.6.1 is working as expected. Thank you for your help.
I think you should add a big warning in the changelogs (CHANGELOG.md & GitHub releases) about v2.6 & Java8.

**Comment by nmuesch** [comment]
Thanks for the troubleshooting steps and details. I've called this out in the v2.6.1 release and changelog that  v2.6.1 resolved a compatibility issue introduced with v2.6. I'll go ahead and close this issue for now. Apologies again for any inconvenience.
